raster-foundry/raster-foundry
# Auto expire stuck tasks

## Overview

This PR automatically expires tasks that are stuck in either `*_IN_PROGRESS` task status.

### Checklist


### Notes

Needed indices:

- [x] task status
- [x] task locked on

~`wip` part is that it doesn't do anything automatic yet, there's just a db function that has tests that show that it works as expected. Tomorrow is the fun part :sunglasses:~

## Testing Instructions

- bring up local annotate pointing to local rf
- assemble backsplash (if you haven't in a while, assemble the api server as well, but it's not technically necessary)
- set DB_TASK_STATUS_EXPIRATION_SECONDS in docker-compose.yml to a nice low value, like 61, so that you'll definitely have time to do stuff after you start labeling a task but won't have to wait too long to see that task expiration was successful
- bring up your rf server
- create a new project
- start labeling one of the tasks
- close your browser
- confirm that the task's status in the db is `LABELING_IN_PROGRESS`
- wait for the next backsplash-server cron to run
- check your task status again, it should be `Unlabeled`

Closes raster-foundry/annotate#507
